### ISSUE TITLE
MemSQL doesn't support the SET @@var = @other_var syntax

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13867]
The workaround introduced in jOOQ 3.15 for MySQL / MariaDB's `GROUP_CONCAT` auto-truncation problem doesn't work in Single Store: https://github.com/jOOQ/jOOQ/issues/12092

While this is possible:

```sql
set @t = @@group_concat_max_len;
```

This doesn't work:

```sql
set @@group_concat_max_len = @t;
```

The error is:

> SQL Error [1064] [42000]: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '@t' at line 1

It appears that this is some sort of parser bug, e.g. this also doesn't work:

```sql
set @@group_concat_max_len = (select @t);
```

The error being:

> SQL Error [1064] [42000]: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '@t)' at line 1

It seems that as soon as a `@@` variable is set, no more `@` can be parsed for the remainder of the statement. I guess we'll just have to leave `@@group_concat_max_len` as it is.

**Comment by lukaseder** [comment]
I've reported this here: https://www.singlestore.com/forum/t/cannot-assign-variable-to-group-concat-max-len/4876

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.3 (https://github.com/jOOQ/jOOQ/issues/13868), and 3.16.9 (https://github.com/jOOQ/jOOQ/issues/13869)
